pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -271,79 +272,102 @@ def circuit(self) -> Circuit:
                     "condition_bits": [cond_reg[k] for k in range(len(cond_reg))],
-                    "condition_value": i.condition[1],
+                    "condition_value": instr.condition[1],
                 }
             optype = None
-            if type(i) == ControlledGate:
-                if type(i.base_gate) == qiskit_gates.RYGate:
+            if type(instr) == ControlledGate:
+                if type(instr.base_gate) == qiskit_gates.RYGate:
                     optype = OpType.CnRy
+                elif type(instr.base_gate) == qiskit_gates.YGate:
+                    optype = OpType.CnY
+                elif type(instr.base_gate) == qiskit_gates.ZGate:
+                    optype = OpType.CnZ
                 else:
-                    # Maybe handle multicontrolled gates in a more general way,
-                    # but for now just do CnRy
-                    raise NotImplementedError(
-                        "qiskit ControlledGate with "
-                        + "base gate {} not implemented".format(i.base_gate)
-                    )
-            elif type(i) == PauliEvolutionGate:
+                    if type(instr.base_gate) in _known_qiskit_gate:
+                        optype = OpType.QControlBox  # QControlBox case handled below
+                    else:
+                        raise NotImplementedError(
+                            f"qiskit ControlledGate with base gate {instr.base_gate}"
+                            + "not implemented"
+                        )
+            elif type(instr) == PauliEvolutionGate:
                 pass  # Special handling below
             else:
-                optype = _known_qiskit_gate[type(i)]
+                optype = _known_qiskit_gate[type(instr)]
             qubits = [self.qbmap[qbit] for qbit in qargs]
             bits = [self.cbmap[bit] for bit in cargs]
 
             if optype == OpType.Unitary2qBox:
-                u = i.to_matrix()
+                u = instr.to_matrix()
                 ubox = Unitary2qBox(u)
                 # Note reversal of qubits, to account for endianness (pytket unitaries
                 # are ILO-BE == DLO-LE; qiskit unitaries are ILO-LE == DLO-BE).
                 self.tkc.add_unitary2qbox(
                     ubox, qubits[1], qubits[0], **condition_kwargs
                 )
-            elif type(i) == PauliEvolutionGate:
-                qpo = _qpo_from_peg(i, qubits)
+            elif optype == OpType.QControlBox:
+                base_tket_gate = _known_qiskit_gate[type(instr.base_gate)]
+                params = [param_to_tk(p) for p in instr.base_gate.params]
+                n_base_qubits = instr.base_gate.num_qubits
+                sub_circ = Circuit(n_base_qubits)
+                # use base gate name for the CircBox (shows in renderer)
+                sub_circ.name = instr.base_gate.name.capitalize()
+                sub_circ.add_gate(base_tket_gate, params, list(range(n_base_qubits)))
```

Comment:
Do you think that adding the `base_tket_gate` to the list given by `list(range(n_base_qubits)` is correct here? I guess I have implicitly assumed that the base gate acts on the base qubits symmetrically.

Reply:
I think it is correct: a qiskit `Gate` (https://qiskit.org/documentation/stubs/qiskit.circuit.Gate.html#qiskit.circuit.Gate) is abstract and doesn't act on specific qubits,